client/my-sites/checkout/upsell-nudge/index.jsx

```diff
@@ -354,7 +354,8 @@ export class UpsellNudge extends Component {
 	getThankYouPageUrlForIncomingCart = ( shouldHideUpsellNudges = true ) => {
 		const getThankYouPageUrlArguments = {
 			siteSlug: this.props.siteSlug,
-			receiptId: this.props.receiptId || 'noPreviousPurchase',
```

Comment:
This overloading of the receiptId greatly confuses the types, and I don't think it's really necessary. I opted instead to add a new argument called `noPurchaseMade` that makes this case explicit.